securedrop/static/js/journalist.js

```diff
@@ -2,93 +2,172 @@
+    });
+  }
 
-  $("#delete_collections").click(function () {
-    var checked = $("ul#cols li :checkbox").filter(":visible").filter(function(index) {
-        return $(this).prop('checked');
+  var deleteCollections = document.getElementById('delete_collections');
+  if (deleteCollections) {
+    deleteCollections.addEventListener('click', function(evt) {
+      var checked = Array.prototype.filter.call(
+        document.querySelectorAll('ul#cols li input[type="checkbox"]:not(.hidden)'),
+        function(obj) { return obj.checked; }
+      );
+      if (checked.length > 0) {
+        var confirmed = confirm("Are you sure you want to delete the " + checked.length + " selected collection"
+                                  + (checked.length > 1 ? "s?" : "?"));
+        if (!confirmed) {
+          evt.preventDefault();
+        }
+        return confirmed;
+      }
+      // Don't submit the form if no collections are selected
+      evt.preventDefault();
+      return false;
     });
-    if (checked.length > 0) {
-      return confirm("Are you sure you want to delete the " + checked.length + " selected collection" + (checked.length > 1 ? "s?" : "?"));
-    }
-    // Don't submit the form if no collections are selected
-    return false;
-  });
+  }
 
-  $("#delete_selected").click(function () {
-      var checked = $("ul#submissions li :checkbox").filter(function() {
-          return $(this).prop('checked')
-      });
+  var deleteSelected = document.getElementById('delete_selected');
+  if (deleteSelected) {
+    deleteSelected.addEventListener('click', function(evt) {
+      var checked = Array.prototype.filter.call(
+        document.querySelectorAll('ul#submissions li input[type="checkbox"]:not(.hidden)'),
+        function(obj) { return obj.checked; }
+      );
       if (checked.length > 0) {
-          return confirmed = confirm("Are you sure you want delete the " + checked.length + " selected submission" + (checked.length > 1 ? "s?" : "?"));
+        var confirmed = confirm("Are you sure you want delete the " + checked.length + " selected submission"
+                                  + (checked.length > 1 ? "s?" : "?"));
+        if (!confirmed) {
+          evt.preventDefault();
+        }
+        return confirmed;
       }
       // Don't submit the form if no submissions are selected
+      evt.preventDefault();
       return false;
-  });
+    });
+  }
 
-  $("#unread a").click(function(){
-    $("#unread").html("unread: 0");
-  });
+  var unread = document.querySelectorAll('.unread a');
+  if (unread) {
+    [].forEach.call(unread, function(el) {
+      el.addEventListener('click', function() {
+        document.querySelector('.unread').innerHTML = "unread: 0";
+      });
+    });
+  }
 
-  var filter_codenames = function(value){
-    if(value == ""){
-      $('ul#cols li').show()
+  var filter_codenames = function(value) {
+    if (value == "") {
+      [].forEach.call(document.querySelectorAll('ul#cols li'), function(el) {
```

Comment:
Is there any reason you're using `[].forEach.call(...)` instead of:

```js
document.querySelectorAll('ul#cols li').forEach(function(el) { ... });
```
?